frontend/src/components/dashboard/noticeOfWorkHomePage/NoticeOfWorkHomePage.js

```diff
@@ -12,20 +16,99 @@ import { getNoticeOfWorkList, getNoticeOfWorkPageData } from "@/selectors/notice
 
 const propTypes = {
   fetchNoticeOfWorkApplications: PropTypes.func.isRequired,
-  // use NoW custom prop once this feature is fully implemented
-  // eslint-disable-next-line react/forbid-prop-types
-  noticeOfWorkApplications: PropTypes.arrayOf(PropTypes.any).isRequired,
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
+  location: PropTypes.shape({ search: PropTypes.string }).isRequired,
+  pageData: CustomPropTypes.pageData.isRequired,
+  noticeOfWorkApplications: PropTypes.arrayOf(CustomPropTypes.nowApplication).isRequired,
 };
 
 export class NoticeOfWorkHomePage extends Component {
-  state = { isLoaded: false };
+  params = queryString.parse(this.props.location.search);
+
+  state = {
+    isLoaded: false,
+    params: {
+      page: Strings.DEFAULT_PAGE,
+      per_page: Strings.DEFAULT_PER_PAGE,
+      ...this.params,
+    },
+  };
 
   componentDidMount() {
-    this.props.fetchNoticeOfWorkApplications().then(() => {
-      this.setState({ isLoaded: true });
-    });
+    const params = this.props.location.search;
+    const parsedParams = queryString.parse(params);
+    const { page = this.state.params.page, per_page = this.state.params.per_page } = parsedParams;
+    if (params) {
+      this.renderDataFromURL();
+    } else {
+      this.props.history.push(
+        router.NOTICE_OF_WORK_APPLICATIONS.dynamicRoute({
+          page,
+          per_page,
+        })
+      );
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged) {
+      this.renderDataFromURL(nextProps);
+    }
   }
 
+  componentWillUnmount() {
+    this.setState({ params: {} });
+  }
+
+  renderDataFromURL = (nextProps) => {
+    const params = nextProps ? nextProps.location.search : this.props.location.search;
+    const parsedParams = queryString.parse(params);
```

Comment:
Wrote the same feedback on Jons PR, copying it here, assuming you both copied this logic from `ContactSearch`: 

the params names are confusing here - nextProps should only be used in componentsWillReceiveProps, as the data passed into this function only needs to be params not all of the available props which `nextProps` implies.

You do a check for params in `componentDidMount`, It would be better to pass params in at that point since params already equal this.props.location.search
Then the `renderDataFromURL` will already accept the proper params and this check will no longer be needed.

Reply:
chatted IRL